server/api.js

```diff
@@ -45,32 +27,54 @@ const getChannelHistory = async (channel, oldest, latest) => {
 	return res.data;
 };
 
+const findReaction = (reactions, userId) => {
+	return reactions.filter((reaction) => reaction.users.includes(userId));
+};
 
-router.get("/user/:userId", async (req, res) => {
-	const userId = req.params.userId;
-	let result = {
-		userName: "",
-		statistics: [],
-	};
-	const now = new Date();
-	const weekly = now.getDay() == 0 ? 6 : now.getDay();
-    const monthly = now.getDate() == 0 ? 30 : now.getDate();
-    let time = req.query.time == "month" ? monthly : weekly;
-	let latest = now / 1000;
-	let oldest = latest - 60 * 60 * 24 * time;
-	time = req.query.time == "month" ? 30 : 7;
-	// console.log("duration=", time, " latest=", latest, " oldest=", oldest);
-	const userInfo = await getUserInfo(userId);
-	const userName = userInfo.ok
-		? userInfo.user.real_name
-		: res.status(400).json("The user not found");
-	result.userName = userName;
-	// console.log(userName);
-	const findReaction = (reactions) => {
-		return reactions.filter((reaction) => reaction.users.includes(userId));
+const getStat = async (userId, oldest,latest) => {
+	latest = latest == undefined? "" : latest;
+	console.log(latest);
+	const data = await getChannelHistory("C027M110K9T", oldest, latest);
+	const messageCount = data.messages.filter(
+		(message) =>
+			(message.user == userId && message.type == "message") ||
+			(message.reply_users && message.reply_users.includes(userId))
+	).length;
+	const reactionCount = data.messages.filter(
+		(message) =>
+			message.reactions && findReaction(message.reactions, userId).length > 0
+	).length;
+	const newCount = {
+		messageCount: messageCount,
+		reactionCount: reactionCount,
 	};
-	let data = await getChannelHistory("C028DN05PUG");
-	// console.log(data);
+	return newCount;
+};
+
+router.get("/user/:userId", async (req,res)=>{
+const userId = req.params.userId;
+let result = {
+	userName: "",
+	statistics: [],
+};
+const userInfo = await getUserInfo(userId);
+const userName = userInfo.ok
+	? userInfo.user.real_name
+	: res.status(400).json("The user not found");
+result.userName = userName;
+const now = new Date()/1000;
+const oldest =now - 60*60*7*24;
```

Comment:
would it be better to save this number as a const variable, e.g.

`const WEEK_IN_SECONDS = 60*60*7*24;`

then use it on lines 66 and 89 (where you have it as 604800000)

Reply:
Oh yes, It's a good idea. I'll do.
 Thank you, Laura.